tests/datasets/test_spacenet.py

```diff
@@ -0,0 +1,72 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+import glob
+import os
+import shutil
+from pathlib import Path
+from typing import Generator
+
+import pytest
+import torch
+from _pytest.monkeypatch import MonkeyPatch
+
+from torchgeo.datasets import Spacenet1
+from torchgeo.transforms import Identity
+
+TEST_DATA_DIR = "tests/data/spacenet"
+
+
+class Dataset:
+    def __init__(self, collection_id: str) -> None:
+        self.collection_id = collection_id
+
+    def download(self, output_dir: str, **kwargs: str) -> None:
+        glob_path = os.path.join(TEST_DATA_DIR, self.collection_id, "*.tar.gz")
+        for tarball in glob.iglob(glob_path):
+            shutil.copy(tarball, output_dir)
+
+
+def fetch(collection_id: str, **kwargs: str) -> Dataset:
+    return Dataset(collection_id)
+
+
+class TestSpacenet1:
+    @pytest.fixture
+    def dataset(
+        self,
+        monkeypatch: Generator[MonkeyPatch, None, None],
+        tmp_path: Path,
+    ) -> Spacenet1:
+        radiant_mlhub = pytest.importorskip("radiant_mlhub", minversion="0.2.1")
+        monkeypatch.setattr(  # type: ignore[attr-defined]
+            radiant_mlhub.Dataset, "fetch", fetch
+        )
+        test_md5 = "829652022c2df4511ee4ae05bc290250"
+        monkeypatch.setattr(Spacenet1, "md5", test_md5)  # type: ignore[attr-defined]
+        root = str(tmp_path)
+        transforms = Identity()
+        return Spacenet1(
+            root,
+            image="8band",
```

Comment:
We'll want to test both 8band and rgb. Can you [parametrize this fixture](https://docs.pytest.org/en/6.2.x/fixture.html#fixture-parametrize)? Let me know if you need help, pytest can be confusing if you aren't already familiar with it.

Reply:
I can parametrize the fixture but I'm a bit stuck at testing the number of bands. 
```
class TestSpacenet1:
    @pytest.fixture(params=["rgb", "8band"])
    def dataset(
        self,
        request,
        monkeypatch: Generator[MonkeyPatch, None, None],
        tmp_path: Path,
    ) -> Spacenet1:
        radiant_mlhub = pytest.importorskip("radiant_mlhub", minversion="0.2.1")
        monkeypatch.setattr(  # type: ignore[attr-defined]
            radiant_mlhub.Dataset, "fetch", fetch
        )
        test_md5 = "829652022c2df4511ee4ae05bc290250"
        monkeypatch.setattr(Spacenet1, "md5", test_md5)  # type: ignore[attr-defined]
        root = str(tmp_path)
        transforms = Identity()
        return Spacenet1(
            root,
            image=request.param,
            transforms=transforms,
            download=True,
            api_key="",
        )

    def test_getitem(self, dataset: Spacenet1) -> None:
        x = dataset[0]
        assert isinstance(x, dict)
        assert isinstance(x["image"], torch.Tensor)
        assert isinstance(x["mask"], torch.Tensor)
        assert x["image"].shape[0] == 3 
```


I think we can use `pytest.mark.parametrize` but I'm not sure we can do that here since we're testing the property of an instance. Any suggestions?

Reply:
Yeah, I don't think you can do that. Fixture parametrization is different than test parametrization for some reason. If you store `image` in `self.image` you can do something like:
```python
if dataset.image == "rgb":
    assert x["image"].shape[0] == 3
else:
    assert x["image"].shape[0] == 8
```

Reply:
Great. I'll go with that then.